StRoot/StFwdTrackMaker/include/Tracker/FwdHit.h

```diff
@@ -0,0 +1,164 @@
+#ifndef FwdHit_h
+#define FwdHit_h
+
+#include "KiTrack/IHit.h"
+#include "KiTrack/ISectorConnector.h"
+#include "KiTrack/ISectorSystem.h"
+#include "KiTrack/KiTrackExceptions.h"
+
+#include <memory>
+#include <set>
+#include <string.h>
+#include <vector>
+
+class StHit;
+
+class FwdSystem : public KiTrack::ISectorSystem {
+  public:
+    FwdSystem(const int ndisks) : KiTrack::ISectorSystem(), _ndisks(ndisks){};
+    ~FwdSystem(){/* */};
+    virtual unsigned int getLayer(int diskid) const throw(KiTrack::OutOfRange) {
+        return diskid;
+    }
+
+    int _ndisks;
+    std::string getInfoOnSector(int sec) const { return "TODO"; }
+};
+FwdSystem *gFwdSystem;
+//_____________________________________________________________________________________________
+
+// small class to store Mc Track information
+class McTrack {
+  public:
+    McTrack() {
+        _pt = -999;
+        _eta = -999;
+        _phi = -999;
+        _q = 0;
+    }
+    McTrack(float pt, float eta = -999, float phi = -999, int q = 0,
+            int start_vertex = -1) {
+        _pt = pt;
+        _eta = eta;
+        _phi = phi;
+        _q = q;
+        _start_vertex = start_vertex;
+    }
+    void addHit(KiTrack::IHit *hit) { hits.push_back(hit); }
+
+    void addFSIHit(KiTrack::IHit *hit) { fsi_hits.push_back(hit); }
+
+    float _pt, _eta, _phi;
+    int _tid, _q, _start_vertex;
+
+    std::vector<KiTrack::IHit *> hits;
+    std::vector<KiTrack::IHit *> fsi_hits;
+};
+
+class FwdHit : public KiTrack::IHit {
+  public:
+    FwdHit(unsigned int id, float x, float y, float z, int vid, int tid,
+           TMatrixDSym covmat, std::shared_ptr<McTrack> mcTrack = nullptr )
+        : KiTrack::IHit() {
+        _id = id;
+        _x = x;
+        _y = y;
+        _z = z;
+        _tid = tid;
+        _vid = vid;
+        _mcTrack = mcTrack;
+        _hit = 0;
+        _covmat.ResizeTo( 3, 3 );
+        _covmat = covmat;
+
+        int _map[] = {0, 0, 0, 0, 0, 1, 2, 0, 0, 3, 4, 5, 6}; // ftsref6a
+
+        if (vid > 0)
+            _sector = _map[vid];
+        else {
+            _sector = abs(vid); // set directly if you want
+            // now set vid back so we retain info on the tru origin of the hit
+            _vid = abs(vid) + 9; // we only use this for sTGC only.  Needs to be
+                                 // cleaner in future.
+        }
+    };
+
+    const KiTrack::ISectorSystem *getSectorSystem() const {
+        return gFwdSystem;
+    } // need to implement
+
+    int _tid; // aka ID truth
+    int _vid;
```

Comment:
Need to correct variable names according to the STAR guidelines. From https://www.star.bnl.gov/public/comp/sofi/guidelines/formatguide.xml#Variable_Names

> Data members (also called instance variables or member variables) are prefixed with m. If the data member is static, prefix the variable with s instead.